ydb/core/tx/columnshard/engines/changes/general_compaction.h

```diff
@@ -28,6 +28,50 @@ class TGeneralCompactColumnEngineChanges: public TCompactColumnEngineChanges {
 public:
     using TBase::TBase;
 
+    class IMemoryPredictor {
+    public:
+        virtual ui64 AddPortion(const std::shared_ptr<TPortionInfo>& portionInfo) = 0;
+        virtual ~IMemoryPredictor() = default;
+    };
+
+    class TMemoryPredictorSimplePolicy: public IMemoryPredictor {
+    private:
+        ui64 SumMemory = 0;
+    public:
+        virtual ui64 AddPortion(const std::shared_ptr<TPortionInfo>& portionInfo) override {
+            for (auto&& i : portionInfo->GetRecords()) {
+                SumMemory += i.BlobRange.Size;
+                SumMemory += i.GetMeta().GetRawBytesVerified();
+            }
+            return SumMemory;
+        }
+    };
+
+    class TMemoryPredictorChunkedPolicy: public IMemoryPredictor {
+    private:
+        ui64 SumMemory = 0;
+        THashMap<ui32, ui64> MaxMemoryByColumnChunk;
+    public:
+        virtual ui64 AddPortion(const std::shared_ptr<TPortionInfo>& portionInfo) override {
+            SumMemory += portionInfo->GetRecordsCount() * 24;
+            for (auto&& i : portionInfo->GetRecords()) {
+                SumMemory += i.BlobRange.Size;
+                SumMemory += i.GetMeta().GetRawBytesVerified();
```

Comment:
We have the same logic in TMemoryPredictorSimplePolicy. In the future, someone might forget to change both places. can we put it in a separate general method?